karma.conf.js

```diff
@@ -124,21 +117,50 @@ module.exports = function (config) {
     addSauceTests(cfg);
   }
 
-  // the MOCHA_UI env var will determine if we're running interface-specific
-  // tests.  since you can only load one at a time, each must be run separately.
-  // each has its own set of acceptance tests and a fixture.
-  // the "bdd" fixture is used by default.
-  var ui = env.MOCHA_UI;
-  if (ui) {
-    if (cfg.sauceLabs) {
-      cfg.sauceLabs.testName = 'Interface "' + ui + '" integration tests';
-    }
-    cfg.files = [
-      'test/browser-fixtures/' + ui + '.fixture.js',
-      'test/interfaces/' + ui + '.spec.js'
-    ];
-  } else if (cfg.sauceLabs) {
-    cfg.sauceLabs.testName = 'Unit Tests';
+  /* the MOCHA_TEST env var will be set for "special" cases of tests.
+   * these may require different interfaces or other setup which make
+   * them unable to be batched w/ the rest.
+   */
+  var MOCHA_TEST = env.MOCHA_TEST;
+  switch (MOCHA_TEST) {
+    case 'bdd':
+    case 'tdd':
+    case 'qunit':
+      if (cfg.sauceLabs) {
+        cfg.sauceLabs.testName =
+          'Interface "' + MOCHA_TEST + '" Integration Tests';
+      }
+      cfg.files = [
+        'test/browser-fixtures/' + MOCHA_TEST + '.fixture.js',
+        'test/interfaces/' + MOCHA_TEST + '.spec.js'
+      ];
+      break;
+
+    case 'esm':
+      // for now we will only run against Chrome to test this.
+      if (cfg.sauceLabs) {
+        cfg.sauceLabs.testName = 'ESM Integration Tests';
+        cfg.browsers = ['chrome@latest'];
+        var launcher = cfg.customLaunchers['chrome@latest'];
+        cfg.customLaunchers = {
+          'chrome@latest': launcher
+        };
+      } else if (!env.TRAVIS) {
+        cfg.browsers = ['Chrome'];
```

Comment:
what other browsers currently support native modules? @mochajs/core

Reply:
in accordance to MDN

> This feature is only just beginning to be implemented in browsers natively at this time. It is implemented in many transpilers, such as TypeScript and Babel, and bundlers such as Rollup and Webpack.

![image](https://user-images.githubusercontent.com/6191849/33932041-8e59bcd8-dffa-11e7-91c0-9116c531665b.png)